Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -23,57 +28,86 @@ final class SchemesGenerator: SchemesGenerating {
 
     /// Default version for generated schemes.
     private static let defaultVersion = "1.3"
-
-    /// Generates the schemes for the project manifest.
-    ///
+    
+    /// Generate schemes for a project.
     /// - Parameters:
-    ///   - project: Project manifest.
-    ///   - generatedProject: Generated Xcode project.
-    /// - Throws: A FatalError if the generation of the schemes fails.
-    func generateTargetSchemes(project: Project, generatedProject: GeneratedProject) throws {
-        /// Generate scheme from manifest
-        try project.schemes.forEach { scheme in
-            try generateScheme(scheme: scheme, project: project, generatedProject: generatedProject)
-        }
-
-        /// Generate scheme for every targets in Project that is not defined in Manifest
-        let buildConfiguration = defaultDebugBuildConfigurationName(in: project)
-        try project.targets.forEach { target in
-
-            if !project.schemes.contains(where: { $0.name == target.name }) {
-                let scheme = Scheme(name: target.name,
-                                    shared: true,
-                                    buildAction: BuildAction(targets: [target.name]),
-                                    testAction: TestAction(targets: [target.name], configurationName: buildConfiguration),
-                                    runAction: RunAction(configurationName: buildConfiguration,
-                                                         executable: target.productName,
-                                                         arguments: Arguments(environment: target.environment)))
-
-                try generateScheme(scheme: scheme,
-                                   project: project,
-                                   generatedProject: generatedProject)
+    ///     - project: Project manifest.
+    ///     - xcprojectPath: Path to project's .xcodeproj.
+    ///     - generatedProject: Generated Project
+    ///     - graph: Tuist graph.
+    func generateProjectSchemes(project: Project,
+                                xcprojectPath: AbsolutePath,
+                                generatedProject: GeneratedProject,
+                                graph: Graphing) throws {
+         /// Generate scheme from manifest
+         try project.schemes.forEach { scheme in
+             try generateScheme(scheme: scheme,
+                                xcPath: xcprojectPath,
+                                path: project.path,
+                                graph: graph,
+                                generatedProjects: [project.path: generatedProject])
+         }
+         /// Generate scheme for every targets in Project that is not defined in Manifest
+         let buildConfiguration = defaultDebugBuildConfigurationName(in: project)
+         try project.targets.forEach { target in
```

Comment:
I appreciate this was the existing code - but perhaps this could be simplified slightly

```swift
let userDefinedSchemes = project.schemes.map(\.name)
let defaultSchemeTargets = project.targets.filter { !userDefinedSchemes.contains($0.name) } 
defaultSchemeTargets.forEach {
  // no big if statement :p ...
}
```
This may help make the code clearer for future features like #458

Reply:
Good point!